### PR TITLE
Remove color space

### LINKED ISSUE #1567: remove color_space dependency
>  hsv<->rgb conversion is simple and short enough to just copy it instead of adding yet another dependency

### PR DESCRIPTION
**Pull request by bim9262**
Fixes #1567
Supersedes #1569

### CODE REVIEW

File: src/themes/color.rs [hunk 1]
```diff
@@ -6,70 +6,80 @@ use std::fmt;
...
+impl Rgba {
+    /// Create a new RGBA color.
     ///
     /// `r`: red component (0 to 255).
     ///
     /// `g`: green component (0 to 255).
     ///
     /// `b`: blue component (0 to 255).
+    ///
+    /// `a`: alpha component (0 to 100).
```
Comment (MaxVerevkin) [inline src/themes/color.rs:27]: I think you meant "0 to 255"

File: src/themes/color.rs [hunk 2]
```diff
@@ -6,70 +6,80 @@ use std::fmt;
 use std::ops::Add;
 use std::str::FromStr;
 
-/// An RGB color (red, green, blue).
+/// An RGBA color (red, green, blue, alpha).
 #[derive(Copy, Clone, Debug, Default)]
-pub struct Rgb {
+pub struct Rgba {
     pub r: f64,
```
Comment (MaxVerevkin) [inline src/themes/color.rs:12]: Do we need RGB components to be floating point numbers?

Comment (bim9262) [reply src/themes/color.rs:12]: We don't have to use `f64`s for `Rgba`, but it does require more casting for `Rgba` <-> `Hsva`. I'll put add an additional commit for review.

File: src/themes/color.rs [hunk 3]
```diff
@@ -1,19 +1,193 @@
...
+    /// Create a new RGBA color from the `hex` value.
+    ///
+    /// ```let cyan = Rgba::from_hex(0xffffff);```
+    pub fn from_hex(hex: u32) -> Self {
+        Self {
+            r: (hex >> 24) as u8,
+            g: (hex >> 16) as u8,
+            b: (hex >> 8) as u8,
+            a: hex as u8,
+        }
```
Comment (MaxVerevkin) [inline src/themes/color.rs:42]:
```suggestion
        let [r, g, b, a] = hex.to_be_bytes();
        Self { r, g, b, a }
```

Comment (bim9262) [reply src/themes/color.rs:42]: @MaxVerevkin, that looks good. I'll amend my commit.

**Comment by MaxVerevkin** [comment]
Thanks!

**Pull request merged**